File: 7. Reverse Integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        a = list(str(x))
        if  a[0] != '-':
            a.reverse()
            stra = "".join(a)
            b = int(stra)
            if b >= 2147483647:
                return 0
            else:
                return b
        else:
            a.reverse()
            a.pop()
            a.insert(0,"-")
            stra = "".join(a)
            b = int(stra)
            if b <= -2147483647:
                return 0
            else:
                return(b)

File: 7. Reverse Integer.cpp
class Solution {
public:
    int reverse(int x) {
        long long y = 0;
        while (x) {
            y = y * 10 + x % 10;
            x /= 10;
        }
        return y < INT_MIN || y > INT_MAX ? 0 : y;
    }
};
